tests/test_threat_intelligence.py

```diff
@@ -99,8 +113,731 @@ def test_check_local_ti_files_for_update(
+    if expected_set_evidence_call:
+        mock_db.set_evidence.assert_called()
+    else:
+        mock_db.set_evidence.assert_not_called()    
+
+def test_is_malicious_url(threatintel_obj, mocker):
+    """
+    Test `is_malicious_url` for correctly handling both malicious and non-malicious URLs.
+    """
+    mock_urlhaus_lookup = mocker.patch.object(threatintel_obj.urlhaus, "urlhaus_lookup")
+    mock_urlhaus_set_evidence = mocker.patch.object(threatintel_obj.urlhaus, "set_evidence_malicious_url")
+    # Test Case 1: Malicious URL
+    mock_urlhaus_lookup.return_value = {"description": "Malicious URL"}
+    threatintel_obj.is_malicious_url(
+        "http://malicious.com", "uid123", "2023-11-28 12:00:00", "192.168.1.1", "profile_10.0.0.1", "timewindow1"
+    )
+    mock_urlhaus_set_evidence.assert_called_once()
+    mock_urlhaus_lookup.reset_mock()
+    mock_urlhaus_set_evidence.reset_mock()
+    # Test Case 2: Non-Malicious URL 
+    mock_urlhaus_lookup.return_value = None  
+    threatintel_obj.is_malicious_url(
+        "http://safe.com", "uid123", "2023-11-28 12:00:00", "192.168.1.1", "profile_10.0.0.1", "timewindow1"
+    )
+    mock_urlhaus_set_evidence.assert_not_called()  
+
+@pytest.mark.parametrize(
+    "msg_data, expected_call",
+    [ # Test Case 1: Malicious CNAME in DNS response
+        (
+            {
+                "profileid": "profile_10.0.0.1",
+                "twid": "timewindow1",
+                "stime": "2023-11-28 12:00:00",
+                "uid": "uid123",
+                "proto": "UDP",  
+                "daddr": "8.8.8.8",
+                "is_dns_response": True,
+                "dns_query": "example.com",
+                "to_lookup": "evil.com",  
+                "type": "domain",
+            },
+            "is_malicious_cname",
+        ),
+        # Test Case 2: Malicious domain in direct query
+        (
+            {
+                "profileid": "profile_10.0.0.1",
+                "twid": "timewindow1",
+                "stime": "2023-11-28 12:00:00",
+                "uid": "uid123",
+                "proto": "TCP",
+                "daddr": "192.168.1.1",
+                "to_lookup": "malicious.com",  
+                "type": "domain", 
+            },
+            "is_malicious_domain",
+        ),
+    ],
+)
+def test_main_domain_lookup(threatintel_obj, mocker, msg_data, expected_call):
+    """
+    Test the `main` function's handling of domain name lookups,
+    covering scenarios with DNS responses and direct domain queries.
+    """
+    mock_is_malicious_cname = mocker.patch.object(
+        threatintel_obj, "is_malicious_cname"
+    )
+    mock_is_malicious_domain = mocker.patch.object(
+        threatintel_obj, "is_malicious_domain"
+    )
+    mock_get_msg = mocker.patch.object(threatintel_obj, "get_msg")
+
+    mock_get_msg.return_value = {
+        "data": json.dumps(msg_data)
+    }
+
+    threatintel_obj.main()
+    if expected_call == "is_malicious_cname":
```

Comment:
always avoid if statements that check test cases if you're using mark parametrize. if statements render  mark.parametrize useless